serialization.go

```diff
@@ -158,53 +172,41 @@ func SendJSONError(w http.ResponseWriter, err error) {
 	}
 
 	w.WriteHeader(status)
-	SendJSON(w, err)
+	_ = SendJSON(w, nil, err)
 }
 
 // SendXML sends a XML response.
 // Declared as a variable to be able to override it for clients that need to customize serialization.
-var SendXML = func(w http.ResponseWriter, r *http.Request, ans any) error {
+var SendXML = func(w http.ResponseWriter, _ *http.Request, ans any) error {
 	w.Header().Set("Content-Type", "application/xml")
 	return xml.NewEncoder(w).Encode(ans)
 }
 
 // SendXMLError sends a XML error response.
 // If the error implements ErrorWithStatus, the status code will be set.
-func SendXMLError(w http.ResponseWriter, r *http.Request, err error) {
+func SendXMLError(w http.ResponseWriter, _ *http.Request, err error) {
 	status := http.StatusInternalServerError
 	var errorStatus ErrorWithStatus
 	if errors.As(err, &errorStatus) {
 		status = errorStatus.StatusCode()
 	}
 
 	w.WriteHeader(status)
-	err = SendXML(w, r, err)
+	err = SendXML(w, nil, err)
 	if err != nil {
 		slog.Error("Cannot serialize returned response to XML", "error", err)
 		_, _ = w.Write([]byte(`{"error":"Cannot serialize returned response to XML"}`))
 	}
 }
 
-func SendHTMLError(ctx context.Context, w http.ResponseWriter, err error) error {
-	status := http.StatusInternalServerError
-	var errorStatus ErrorWithStatus
-	if errors.As(err, &errorStatus) {
-		status = errorStatus.StatusCode()
-	}
-
-	w.Header().Set("Content-Type", "text/html; charset=utf-8")
-	w.WriteHeader(status)
-	return SendHTML(ctx, w, err.Error())
-}
-
 // SendHTML sends a HTML response.
 // Declared as a variable to be able to override it for clients that need to customize serialization.
-var SendHTML = func(ctx context.Context, w http.ResponseWriter, ans any) error {
+var SendHTML = func(w http.ResponseWriter, r *http.Request, ans any) error {
 	w.Header().Set("Content-Type", "text/html; charset=utf-8")
 
 	ctxRenderer, ok := any(ans).(CtxRenderer)
 	if ok {
-		return ctxRenderer.Render(ctx, w)
+		return ctxRenderer.Render(r.Context(), w)
```

Comment:
Indeed, sorry that was weird from my part, and moreover fuego's context is equal to r.Context, so I really don't know why I've done this